compiler/rustc_span/src/source_map.rs

```diff
@@ -380,10 +343,10 @@ impl SourceMap {
             nc.pos = nc.pos + start_pos;
         }
 
+        let (filename, _) = self.path_mapping.map_filename_prefix(&filename);
```

Comment:
What I said above https://github.com/rust-lang/rust/pull/83813#discussion_r608585013 was wrong. To stop double remapping metadata embedded in other crates we just need to remove this line.